src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -434,5 +435,18 @@ internal static void SetCurrentUICulture(CultureInfo cultureInfo)
                 Thread.CurrentUICulture.SetValue(thread, cultureInfo);
             }
         }
+
+        internal static class Assembly
+        {
+            private const string TypeName = "System.Reflection.Assembly";
+
+            internal static readonly Type Type = ReflectionUtil.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            internal static readonly MethodInfo GetTypeThrowOnErrorIgnoreCase = Type
+                .GetTypeInfo()
+                .GetDeclaredMethod("GetType", typeof(string), typeof(bool), typeof(bool));
```

Comment:
Can we use CreateDelegate here to make this a direct call to GetType?  That would avoid a TargetInvocattionException on error.